### ISSUE TITLE
Make a combination unavailable for sale

### ISSUE THREAD

**Issue by yannicka** [PrestaShop/PrestaShop#22970]
#### Is your feature request related to a problem?

I would like to be able to continue to display a combination, but to make it unavailable for sale, even though there is still stock.

#### Describe the solution you'd like

An "Available for sale" checkbox, checked by default, which can be unchecked, or something similar.

---
Edit: 

![image](https://user-images.githubusercontent.com/23308427/105846029-e2a6f080-5fdb-11eb-9ad2-167e0398e422.png)

**Comment by hibatallahAouadni** [comment]
Hello @yannicka 

Thanks for you suggestion!
So, what you suggest is that when you create a specific price for a product with combinations, you can eliminate some combinations from the sale, is that so?

Thanks!

**Comment by yannicka** [comment]
No, no specific prices. Let's imagine that I have a product declined in three colors Red, Green and Blue, I want to be able to keep the Green combination in the database, but no longer offer it for sale.

**Comment by hibatallahAouadni** [comment]
Hello @yannicka 

I see :slightly_smiling_face: you mean that you want to add an attribute "Status" to each combination, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/105846029-e2a6f080-5fdb-11eb-9ad2-167e0398e422.png)

Is that so? Please check and feedback.
Thanks!

**Comment by yannicka** [comment]
Yes that's what I need.

**Comment by hibatallahAouadni** [comment]
Hello @yannicka 

Great! Thanks for your feedback.
The Product team will take a look. Please stay tuned!

Ping @PrestaShop/product-team I think it will be useful to add a new attribute "Status" for each combination, do you agree with me?

**Comment by LouiseBonnard** [comment]
Hello @hibatallahAouadni, hi @yannicka, thanks for this issue! @MatShir (and @Julievrz for the use case), it might interest you. ;-)

**Comment by Hlavtox** [comment]
Ping @PrestaShop/product-council This is an extremely often requested feature. When we want to just forever discontinue a combination, we have to delete it. No other way to do this.

**Comment by boo-code** [comment]
Checked on 9.2: there is no per-combination availability flag. `ps_product_attribute` carries `id_product_attribute`, `reference`, `supplier_reference`, `ean13`, `isbn`, `upc`, `mpn`, `wholesale_price`, `price`, `ecotax`, `weight`, `unit_price_impact`, `default_on`, `minimal_quantity`, `low_stock_threshold`, `low_stock_alert` and `available_date`. `available_for_order` exists only on the product.

`available_date` is the closest thing and it does not answer this: it postpones availability to a date rather than withdrawing it.

The existing route is to set the combination's stock to 0 with out-of-stock ordering denied, which does make it unbuyable while leaving it visible. The request explicitly rejects that, because it destroys the stock figure the merchant wants to keep.

So the gap is real, and what it needs before code is a decision on what "unavailable for sale" means for the front office, since the existing product-level flags each answer it differently: `available_for_order` keeps the product visible and shows no add-to-cart, `active` removes it entirely, and a zero stock with ordering denied shows it as out of stock. A combination-level flag that behaves like the first of those is the smallest addition, but it is a product decision rather than a technical one, which is presumably why this has sat in `Waiting for PM` since 2021.
